### ISSUE TITLE
Boot: Fix relocation problem

### ISSUE THREAD

**Issue by phip1611** [georglauterbach/uncore#16]

**Comment by phip1611** [comment]
I've got new information on that: This should be the responsibility of the kernel instead of the bootloader if you want to support ACPI sleep/resume. If the CPU resumes from sleep, all APs need to be initialized again from 16 bit to 64-bit long mode.

**Comment by georglauterbach** [comment]
> I've got new information on that: This should be the responsibility of the kernel instead of the bootloader if you want to support ACPI sleep/resume. If the CPU resumes from sleep, all APs need to be initialized again from 16 bit to 64-bit long mode.
> 
> I don't see the reason for a custom EFI loader here too. Multiboot2 should be fine and the Multiboot2 kernel can bootstrap everything.

The kernel code does not bother with multiboot2 anymore - this is part of the [`bootloader`](https://github.com/rust-osdev/bootloader) :D The kernel EFI loader is therefore solved as well. I don't want to bother with low-level boot code anymore to be honest, as this is mix of concerns (bootloader and kernel code). This should ultimately not be a problem though as the kernel should be capable of handling this nevertheless, right?

**Comment by phip1611** [comment]
> This should ultimately not be a problem though as the kernel should be capable of handling APs, etc. nevertheless, right?

Yes, correct. The ultimate solution would be to have a bootable kernel that boots as an EFI application or via Multiboot2 or via legacy boot to support every possible x86_64 machine :D

**Comment by phip1611** [comment]
I have some ideas and hopefully I can work on this issue soon in my own OS project. If I have anything that works, I write down the relevant pointers here. Until then, I unassigned me for now, however, to clean up what's important in the next weeks.

**Comment by georglauterbach** [comment]
I am currently working on a re-implementation, for RISC-V only at the moment. Hence, I will close this as x86 is not a priority anymore. Feel free to fork the current state; it will take some time before I can merge my current work into `master` :)
